### DISCUSSION TITLE
Custom icon in Messagebox class

### DISCUSSION THREAD

**Discussion by lupon1** [israel-dryer/ttkbootstrap#134] [Q&A] [answered]
Hi everyone!
First of all I just really to thank @israel-dryer for his great work! I love ttkbootstrap!
I have a question about the window's icon in the Messagebox class. Actually, when I call any type of Messagebox window, the rendering use the standard Tk icon.
It's any way to use a custom icon, like the main window one or simply have a message box without icon?

Thanks so much!
Regards, Francesco.

**Comment by israel-dryer** [comment]
yes, you can create a custom `MessageDialog`. Currently, the 'icon' parameter only accepts data rather than a file name or PhotoImage object. I'll see if I can fix this in the future. This is how you would add a custom icon to the dialog.

https://ttkbootstrap.readthedocs.io/en/latest/api/dialogs/messagedialog/

```python
import ttkbootstrap as ttk
from ttkbootstrap.dialogs import MessageDialog
import pathlib

app = ttk.Window()

imgfile = pathlib.Path(__file__).parent / 'study.png'
imgbytes = imgfile.read_bytes()

md = MessageDialog('This is my message', icon=imgbytes)
md.show()

app.mainloop()
```
<img width="200" alt="python_m0dXUassCw" src="https://user-images.githubusercontent.com/42584742/149023685-5a8c1364-d3eb-48dc-bb0d-e10be24a0eac.png">

To build a MessageDialog with no icon, simply exclude the icon parameter.
```python
import ttkbootstrap as ttk
from ttkbootstrap.dialogs import MessageDialog

app = ttk.Window()

md = MessageDialog('This is my message')
md.show()

app.mainloop()
```
<img width="200" alt="python_oNi25mUU5T" src="https://user-images.githubusercontent.com/42584742/149023913-c4dd8883-4540-4be3-b345-ed27b79c5dba.png">

**Reply by lupon1** [reply]
Thanks for your reply but maybe I haven’t explained correctly. I mean the title bar icon. A picture is worth a thousand words...

![image](https://user-images.githubusercontent.com/49123184/149144750-72e4ccff-f323-4137-96e2-45192239afba.png)

This is the code that I have used:

```
import tkinter as tk
import ttkbootstrap as ttk
from ttkbootstrap.dialogs import Messagebox
from os import path

def click_button():
    Messagebox.show_info('Hi, this is a messagebox window!', 'Info')

actualFolder = path.dirname(__file__)
iconFile = path.join(actualFolder, 'files', 'favicon.ico')
win = tk.Tk()
win.iconbitmap(iconFile)
lbl = ttk.Label(win, text='The main window').pack(padx=50, pady=20)
btn = ttk.Button(win, text='Click', command=click_button).pack(padx=50, pady=10)
win.mainloop()
```

**Reply by israel-dryer** [reply]
If you use an ico file, to set the default icon for all windows use `win.iconbitmap(default=iconFile)`

**Comment by israel-dryer** [comment, marked as answer]
@lupon1, this should now work as expected with a fix in #137.

```python
# set default icon using photoimage
root.iconphoto(1, image)

# remove the default icon
root.iconphoto(1, '')

# set the icon for this window only
root.iconphoto(0, image)

# set the default bitmap (.ico)
root.iconbitmap(default=bitmap)

# remove the default bitmap
root.iconbitmap(default='')

# set the bitmap for this window only
root.iconbitmap(bitmap)
```

**Reply by littlejew2000** [reply]
Hi Israel, 

I have set the icon by _root.iconbitmap(default=bitmap)_, 

it works for ttk.Window, but if I apply this to a customized dialog , it will not wok, please refer below snapshot, if you can help ? thanks 

![image](https://user-images.githubusercontent.com/32632972/177189437-bb053d63-43da-4ed4-94a8-f18a108bfc29.png)

![image](https://user-images.githubusercontent.com/32632972/177188899-8778b79a-d1a8-468b-bad7-d81f4c134bd4.png)

**Comment by littlejew2000** [comment]
Hi Israel,

I found a way to change the icon of a customized dialog,  but need to change ttkbootstrap\dialogs source,
for example:

![image](https://user-images.githubusercontent.com/32632972/177192033-d011c7ca-5103-44f9-8d43-3b64d57d7a9d.png)

it works for customized dialog
![image](https://user-images.githubusercontent.com/32632972/177192239-a96e9d6f-3a28-4222-8018-ca77efdaa331.png)

**Reply by oochain** [reply]
The following code works for me.
Have to set `root.iconbitmap(bitmap=favicon_ico_path)` and `root.iconbitmap(bitmap=favicon_ico_path)` separately.
Not sure why I cannot put them into one line.
 `root.iconbitmap(bitmap=favicon_ico_path, default=favicon_ico_path)` won't work.

![image](https://user-images.githubusercontent.com/13476287/183863604-5275245b-aa2c-46b9-aeb0-a53180c57b38.png)

**Reply by michael-hoang** [reply]
thank you! this worked perfectly!
